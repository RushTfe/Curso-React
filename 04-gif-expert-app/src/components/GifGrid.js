import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {


  const [images, setImages] = useState([]);

  useEffect( () => getGifs(), []);


    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=onepunch&limit=10&api_key=7mmZNL56QHm56pSMJqhgxSDWNsXia1B1'
        const resp = await fetch ( url );

        const { data } = await resp.json();
        
        const gifs = data.map(  img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs);

        setImages( gifs );
    
    }

  return (
	  <>
		<h3>{ category }</h3>
		<div className='card-grid'>

			{
				images.map( img => (
					<GifGridItem  
						key={ img.id }
						{ ...img } 
					/>
				))
			}
		</div>
	</>

  )
}

const getImagesInfo = (images) => images.map(({id, title}) => <li key={id}>{title}</li>);