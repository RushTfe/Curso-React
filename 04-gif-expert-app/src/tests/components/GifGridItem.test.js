import { GifGridItem } from '../../components/GifGridItem';
import React from 'react';
import { shallow } from 'enzyme'

describe('Pruebas sobre Gif Expert', () => {
    let wrapper;
    const title = 'Titulo del item';
    const url = 'https://localhost/imagen.jpg'

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={ title } url={ url } />);
    })
    
    test('Debe mostrar el componente correctamente', () => { 
        
        expect( wrapper ).toMatchSnapshot();

     })
})

