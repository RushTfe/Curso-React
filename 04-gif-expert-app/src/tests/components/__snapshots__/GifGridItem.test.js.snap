// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre Gif Expert Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo del item"
    src="https://localhost/imagen.jpg"
  />
  <p>
    Titulo del item
  </p>
</div>
`;
